### DISCUSSION TITLE
anonymity

### DISCUSSION THREAD

**Discussion by phnoel** [aclap-dev/video-downloadhelper#448] [Help and Q&A]
### Your question

Hello,
I am logged in (login + password) on a training site that offers videos.
I use a VPN.
Is downloading videos with Videodownload helper perfectly anonymous?

Thanks

### Addon & CoApp version and details

version: 8.2.2.8
channel: stable
build date: 2024-03-18
build options: linuxlic: false, noyt: false, browser: firefox
lang: fr
license: unset
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:125.0) Gecko/20100101 Firefox/125.0
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.19",
  "binary": "C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "windows",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "C:\\Users\\phili",
  "converterBinary": "C:\\Program Files\\DownloadHelper CoApp\\ffmpeg.exe",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "c30f360955"
}

**Comment by jcfrog** [comment]
Hi,
One thing is for sure, VDH does not track you.

**Reply by phnoel** [reply]
And the site owner? Will he see me download the videos?

**Reply by GrampaWildWilly** [reply]
I do not work for VDH.  I'm just another user.

The difference between playing a video in a web page & downloading the same video is the speed.  When you just sit & watch the thing, it goes by at 24fps or 30fps or whatever the author of the video recorded/edited it at.  That usually shows up as bursts of data transfer, maybe at a few hundred thousand bytes per second, whatever the web site provides.  But they're bursts of data transfer.  That's what HLS does.  It lets the web site serve you a chunk of the video, enough for it to play smoothly in your browser.  When you are nearly done watching the current chunk, they will download the next chunk, just one chunk, & it will look to you like everything is just playing.  They don't send you the whole video.  They send you only enough for you to watch at the moment.  If you pause the video or stop it altogether, no excess data has been transferred to your system.  You can walk away.  You won't care.   They won't care.  That's why they chunk the content.  That's how HLS works.

When you download something, there's no bursts.  I'll get to a caveat to that in a moment.  But a download isn't like playing back.  The download runs at a steady speed.  That speed is usually determined by the web site.  Most web sites won't let you download at 500Mbs or whatever you're paying your ISP for.  Most web sites know they have to provide service to many users concurrently, so they aren't going to let any single user monopolize their service.  Instead, this is the caveat, the web site throttles the data transmitted to you.  Let's say, for argument's sake, they give you 1 or 2 million bytes per second.  That's a small fraction of your available bandwidth but the web site is protecting their service.  In order to give you 1 or 2 million bytes per second AVERAGE download speed, they are probably sending you a certain amount at maybe 50 million bytes per second & then they pause.  Observe line behavior in your Resource Monitor . . . or whatever equivalent you have on your system.  It is very educational.

About line speed.  Your ISP may be telling you they are giving you 500M service.  That's somewhat deceptive.  First off, it's not meg.  It's million.  There's a difference between meg & million.  Meg is 1024x1024x1024 which is more than 1 million.  They aren't giving you meg.  They're giving you million.  Plus it's bits.  Not bytes.  Bits.  Nobody thinks in bits.  But it sounds sexier to glibly say they're giving you 500 meg.  No.  It's 500/8 which is 62.5 million bytes per second.  Not 500M.  62.5 million bytes per second.  They're trying to dazzle you with smoke & mirrors.  Be aware.

So yes, if the web site bothered to pay attention, they could distinguish somebody playing things back from somebody downloading them.  In practice, web sites don't bother with that.  They just configure their servers to provide whatever service they're going to provide & let them run more or less unattended.

But why would you care if the site knows you're downloading from them?  You're logged in there.  They have already identified you.  If you successfully download from there, that is an admission on the part of the web site that you are allowed to do that.  If they don't want people downloading their content, they will protect it with DRM.

In summary: Don't worry.  Be happy.

**Reply by phnoel** [reply]
It's clear! Thanks a lot!
